liferay-echo/liferay-portal
# LPS-184922 Refine the process of importing fragments

Motivation
=======
We want to improve the process of importing fragments.
[Link to story or ticket](https://issues.liferay.com/browse/LPS-184922)

Proposed Solution
========
We have created a new modal for importing in React so that we can customize better its UX. We have also move the warning messages to the view.jsp so that they are visible after closing the modal.
There will be follow up PRs changing the alert messages and the importation of page templates.

Steps to Verify:
----------------
1. Add `feature.flag.LPS-174939=true` to your portal-ext.properties
2. Go to Design -> Fragments and in the three dots menu click on Import (everything should work the same in importing fragment sets and fragments)
3. Check that the modal width is 600px.
4. Check that there is a paragraph before the input with the text: “Select a Zip File containing one or multiple entries.”.
5. Check that the "Select file" label has been replaced with with "File".
6. The "Overwrite Existing Entries" label should not be bold.
7. The import button should be disabled until a zip file is selected.
8. Try importing fragments with no errors and with some errors. Check that the success, error and warning messages appear outside the modal and after the modal has been closed.
9. Check that everything works without the Feature Flag

Screenshoots:
----------------
![Captura de pantalla 2023-05-18 a las 13 06 25](https://github.com/liferay-echo/liferay-portal/assets/3276218/2e9be493-1cf3-44ae-9e58-b9953ee92100)
